okapi-core/src/test/java/org/folio/okapi/service/impl/DockerModuleHandleTest.java

```diff
@@ -136,6 +150,124 @@ private void dockerMockHandle(RoutingContext ctx) {
     }
   }
 
+  @Test
+  public void testDockerPull(TestContext context) {
+    Vertx vertx = Vertx.vertx();
+    int dockerPort = 9231;
+
+    Router router = Router.router(vertx);
+    router.routeWithRegex("/.*").handler(this::dockerMockHandle);
+
+    HttpServerOptions so = new HttpServerOptions().setHandle100ContinueAutomatically(true);
+    HttpServer listen = vertx.createHttpServer(so)
+        .requestHandler(router)
+        .listen(dockerPort, context.asyncAssertSuccess());
+    LaunchDescriptor ld = new LaunchDescriptor();
+    ld.setWaitIterations(2);
+    ld.setDockerImage("folioci/mod-x");
+    ld.setDockerPull(true);
+
+    Ports ports = new Ports(9232, 9233);
+    JsonObject conf = new JsonObject().put("dockerUrl", "tcp://localhost:" + dockerPort);
+
+    dockerPullJson = new JsonObject().put("message", "some message");
+    dockerPullStatus = 200;
+
+    {
+      DockerModuleHandle dh = new DockerModuleHandle(vertx, ld,
+          "mod-users-5.0.0-SNAPSHOT", ports, "localhost",
+          9231, conf);
+      Async async = context.async();
+      dh.pullImage().onComplete(context.asyncAssertSuccess(x -> {
+        async.complete();
+      }));
+      async.await();
+    }
```

Comment:
All these code blocks are almost the same and might be extracted into a method, for example

```
boolean pullImage(TestContext context, Vertx vertx, String host, JsonObject conf) {
  LaunchDescriptor ld = new LaunchDescriptor();
  ld.setWaitIterations(2);
  ld.setDockerImage("folioci/mod-x");
  ld.setDockerPull(true);
  Ports ports = new Ports(9232, 9233);
  DockerModuleHandle dh = new DockerModuleHandle(vertx, ld,
          "mod-users-5.0.0-SNAPSHOT", ports, host,
          9231, conf);
  Async async = context.async();
  AtomicBoolean succeeded = new AtomicBoolean();
  dh.pullImage().onComplete(done -> {
    succeeded.set(done.succeeded()));
    async.complete();
  });
  async.await();
  return succeeded.get();
}
```
```
assertThat(pullImage(context, vertx, "localhost", conf), is(true));
```